src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/MacRazorProjectHostBase.cs

```diff
@@ -264,15 +256,12 @@ protected void RemoveDocument(HostProject hostProject, string filePath)
         return true;
     }
 
-    private static string? ResolveFallbackIntermediateOutputPath(IMSBuildEvaluatedPropertyCollection projectProperties, string intermediateOutputPathValue)
+    private string? ResolveFallbackIntermediateOutputPath(IMSBuildEvaluatedPropertyCollection projectProperties, string intermediateOutputPathValue)
     {
-        if (!projectProperties.HasProperty(MSBuildProjectDirectoryPropertyName))
-        {
-            // Can't resolve the project, bail.
-            return null;
-        }
+        var projectDirectory = projectProperties.HasProperty(MSBuildProjectDirectoryPropertyName)
+            ? projectProperties.GetValue(MSBuildProjectDirectoryPropertyName)
+            : DotNetProject.BaseDirectory.ToString();
```

Comment:
This is belt and braces.. the current code is failing because `MSBuildProjectDirectory` is null for MAUI projects on Mac, so falling back to the `BaseDirectory` makes sense, but also the above change means it won't get here.